alf/layers.py

```diff
@@ -100,8 +109,41 @@ def unflatten(self, tensor):
                      tuple(tensor.shape[1:])))
 
 
+class Module(nn.Module):
+    """Alf's nn.Module wrapper"""
```

Comment:
What's the purpose of having this new Module class?

Reply:
This is to replace the nn.Module in alf for all custom code we need to have.  Here, it tracks the mode of rollout vs replay (during training).  It is also responsible for generating the mode string for tensorboard plotting purposes.

Reply:
Generally let's be conservative about changing some very basic module or building block because it will usually bring in efficiency issues or code structure issues. If there is some other way like common.mark_training to achieve the same purpose them we should try to avoid this.